### PR TITLE
Support state output

### PR DESCRIPTION
**Pull request by cbush**
Below are some examples of changes when running on realm tutorial codebase. Note that the changes are basically whitespace and newlines at end of file:

Generate tutorial snippets:
```shell
$ bluehawk -s swift-ios/ -d ~/docs/docs-realm/source/tutorial/generated/code/final  --state final --snippets
$ git diff
```
image

Generate tutorial repo branch:
```shell
$ bluehawk -s swift-ios/ -d ~/tutorial/realm-tutorial-ios-swift --state start
$ git diff
```
image

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -13,8 +15,9 @@ async function run(): Promise<void> {
     .command("--source", "The file or folder to process")
     .command("--destination", "The output folder")
     .command("--ignores", "A glob list of patterns to ignore")
+    .command("--state", "Which state to render")
```
Comment (terakilobyte) [inline src/index.ts:18]: thought: Using a flag her to signify which state is an interesting design choice that may warrant revamp in the future.

Passing the desired state requires knowing the contained states a priori. Perhaps a better method would be to somehow find states and convert the ``state`` flag to be exclusionary?

**Review by terakilobyte — APPROVED** [review]
lgtm

File: src/index.ts (hunk 1 above)
Comment (cbush) [reply src/index.ts:18]: Curious, why wouldn't the user know which states are in their codebase? They are marking up the codebase themselves.

That said, I like the idea of a command that prints a list of available states in a project.

Comment (terakilobyte) [reply src/index.ts:18]: To be fair, something like this works

```sh
rg --no-filename -i 'state-start' . | awk 'NF>1{print $NF}' | sort -u
```

It just seems like the tool should output both states in one run, but we'll have to see what typical usage is!

**Pull request merged**